tests/e2e/run-e2e-tests.sh

```
@@ -0,0 +1,458 @@
+# Main test execution
+main() {
+    log_info "Starting quickshell-polkit-agent E2E tests"
+    log_info "=========================================="
+
+    # Create test results directory
+    mkdir -p "$TEST_RESULTS_DIR"
+
+    # Start D-Bus session bus for polkit agent registration
+    if [ -z "$DBUS_SESSION_BUS_ADDRESS" ]; then
+        log_info "Starting D-Bus session bus..."
+        eval $(dbus-launch --sh-syntax)
```

Shellcheck (0.11.0)

[warning] Quote this to prevent word splitting.

(SC2046)